web/api/v1/api.go

```diff
@@ -127,6 +130,59 @@ func (api *API) options(r *http.Request) (interface{}, *apiError) {
 	return nil, nil
 }
 
+// TODO(jsanford): FIXME: This is a sample implementation.
+func (api *API) queryRule(r *http.Request) (interface{}, *apiError) {
+	var ts model.Time
+	if t := r.FormValue("time"); t != "" {
+		var err error
+		ts, err = parseTime(t)
+		if err != nil {
+			return nil, &apiError{errorBadData, err}
+		}
+	} else {
+		ts = api.now()
+	}
+
+	st, err := promql.ParseStmts(string(r.FormValue("query")))
+	if err != nil {
+		return nil, &apiError{errorBadData, err}
+	}
+
+	rls, err := rules.NewRules(st)
+	if err != nil {
+		return nil, &apiError{errorBadData, err}
+	}
+
+	var v []interface{}
+	for _, r := range rls {
+		vector, err := r.Eval(ts, api.QueryEngine)
+		if err != nil {
+			switch err.(type) {
+			case promql.ErrQueryCanceled:
+				return nil, &apiError{errorCanceled, err}
+			case promql.ErrQueryTimeout:
+				return nil, &apiError{errorTimeout, err}
+			}
+			return nil, &apiError{errorExec, err}
+		}
+
+		// TODO(jsanford): Do we want to return this at all?  Probably not.
```

Comment:
Oh!  That's a great idea.

I updated this PR with a better query_rule that accepts some arguments:

end (when the rules should be evaluated, defaults to now)
count (how many evaluations we should do, defaults to "1")
step (duration between evaluations, defaults to 60 seconds)

Now, you can actually test full ALERT rules with FOR!  You can set "count=10" and it will evaluate the rules as if they had been run over the past 10 minutes.  You can also set "end=TIME" to do the rule test at a time in the past.